### PR TITLE
Use nrf_cloud lib's new encode/decode functionality for nRF Cloud data 

### PR DESCRIPTION
**Pull request by jayteemo**
Use features from https://github.com/nrfconnect/sdk-nrf/pull/11169 in apps/samples.
IRIS-6060

### CODE REVIEW

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 1]
```diff
@@ -289,7 +289,7 @@ This plot is useful for tracking, visualizing, and debugging connection loss, re
 Device message formatting
 =========================
 
-This sample constructs JSON-based `device messages <nRF Cloud Device Messages_>`_ using `cJSON`_.
+This sample constructs JSON-based `device messages <nRF Cloud Device Messages_>`_ using.
```
Comment (plskeggs) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:292]: Forgot to remove " using" here.

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c [hunk 2]
```diff
@@ -150,22 +151,17 @@ static int send_gnss(const struct location_event_data * const loc_gnss)
...
+	ret = nrf_cloud_obj_gnss_msg_create(&msg_obj, &gnss_pvt);
 
 	if (ret == 0) {
 		/* Send the location message */
-		ret = send_device_message_cJSON(msg_obj);
-	}
-
-	if (msg_obj) {
-		cJSON_Delete(msg_obj);
+		ret = send_device_message(&msg_obj);
```
Comment (plskeggs) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c:164]: do you need to call `nrf_cloud_obj_free()`?

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c [hunk 3]
```diff
@@ -267,32 +258,33 @@ static void handle_at_cmd_requests(const char *const msg)
...
 	}
 
-	/* Free the old container object and create a new one to contain our response */
-	cJSON_Delete(msg_obj);
-	msg_obj = create_timestamped_device_message(
-		NRF_CLOUD_JSON_APPID_VAL_MODEM,
-		NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA
-	);
+	/* Free and reset the old container object and create a new one to contain our response */
+	(void)nrf_cloud_obj_free(&msg_obj);
```
Comment (plskeggs) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c:262]: Is `cmd` no longer valid after this?  If so, set it to `NULL`?

**Review by plskeggs — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by trantanen — APPROVED** [review]
MOSH part looks good

**Review by glarsennordic — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by VTPeltoketo — APPROVED** [review]

**Pull request merged**